nca/NetworkConfig/NetworkConfig.py

```diff
@@ -265,6 +268,28 @@ def allowed_connections(self, from_peer, to_peer, layer_name=None):
 
         return allowed_conns_res, captured_flag_res, allowed_captured_conns_res, denied_conns_res
 
+    def allowed_connections_optimized(self, layer_name=None):
+        """
+        Computes the set of allowed connections between any relevant peers.
+        :param NetworkLayerName layer_name: The name of the layer to use, if requested to use a specific layer only
+        :return: allowed_conns: all allowed connections for relevant peers.
+        :rtype: ConnectivityProperties
+        """
+        if layer_name is not None:
+            if layer_name not in self.policies_container.layers:
+                conns_res = self.policies_container.layers.empty_layer_allowed_connections_optimized(self.peer_container,
+                                                                                                     layer_name)
+            else:
+                conns_res = self.policies_container.layers[layer_name].allowed_connections_optimized(self.peer_container)
+        else:
+            conns_res = ConnectivityProperties.make_all_props()  # all connections
+            for layer, layer_obj in self.policies_container.layers.items():
+                conns_per_layer = layer_obj.allowed_connections_optimized(self.peer_container)
+                # all allowed connections: intersection of all allowed connections from all layers
+                conns_res &= conns_per_layer
+
+        return conns_res
```

Comment:
should also return allowed captured connections, denied connections, captured flag -- as preparation for next queries to handle.

Reply:
Done